compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala

```diff
@@ -621,13 +621,18 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
     private def reportCannotDisambiguateError(alts: List[Exported]): Unit = {
       val currentClass = currentClassSym.get
 
-      // Find a position that is in the current class for decent error reporting
-      val pos = alts.collectFirst {
+      /* Find a position that is in the current class for decent error reporting.
+       * If there are more than one, always use the "highest" one (i.e., the
+       * one coming last in the source text, so that we reliably display the
+       * same error in all compilers.
```

Comment:
I just realized the opening `(` is not closed. You might as well fix it at the same time as adapting the test.